packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3291,12 +3336,65 @@ class _NavigatorReplaceObservation extends _NavigatorObservation {
   }
 }
 
+/// A ChangeNotifier that notifies when its List of _RouteEntries is mutated.
+class _History extends ChangeNotifier {
+  final List<_RouteEntry> _value = <_RouteEntry>[];
+
+  /// Return a copy of the inner value.
+  ///
+  /// This allows read-only access to the value.
+  List<_RouteEntry> get value => List<_RouteEntry>.from(_value);
```

Comment:
This prevents mutations that don't notify the listeners.  If I returned `_value` directly, I would be able to do this:

```dart
_history.value.add(element);
```

Which would cause a mutation without notifying any listeners.  I'll clarify in these docs a bit more.

That's the reason why I created this _History class, really.  I wanted a clean way to be notified of when the route list changed.  Let me know if there's a better way to do this!

Reply:
I was able to do this by extending Iterable and implementing Iterator, which I think ended up being pretty slick, thanks for the suggestions.